pyproject.toml

```diff
@@ -1,8 +1,25 @@
 [tool.mypy]
-warn_unused_configs = true
 python_version = "3.10"
-mypy_path = "src/py"
+mypy_path = ["src/py", "pyodide-build"]
+show_error_codes = true
+warn_unreachable = true
+enable_error_code = ["ignore-without-code", "redundant-expr", "truthy-bool"]
+
+# Strict checks
+warn_unused_configs = true
 check_untyped_defs = true
+disallow_any_generics = false
+disallow_subclassing_any = false
+disallow_untyped_calls = false
+disallow_untyped_defs = false
+disallow_incomplete_defs = false
+disallow_untyped_decorators = false
+no_implicit_optional = true
+warn_redundant_casts = true
+warn_unused_ignores = true
+warn_return_any = false
+no_implicit_reexport = false
+strict_equality = false
```

Comment:
I'm not really sure about maintaining a custom set of mypy flags or figuring out what each of these do.  Using default flags, with maybe a strict mode would be preferable IMO.

I agree that mypy doesn't catch everything, but it will also not catch everything with a custom configuration.

Also it doesn't look like this required much changes in the code aside from the  `no_implicit_optional`

Reply:
This is copied from strict=yrue; the idea is we can slowly work on making all these true. Once they are all true, we can replace all of them with strict=true.

By default, everything is false, so even just a single true in a custom configuration is better than nothing.

I intentionally only enabled things that were easy to enable in a first step (that is, didn't require lots of adding or fixing types).

Reply:
For better or worse, this is the design of mypy; everything off by default, and then turning one what you'd like to be strict about. `strict=true / --strict` was a concession, which is why it wasn't even available in the config mode for a long time (and still doesn't get handled properly in config mode if you try to override anything).

Reply:
If you prefer, I can replace the `= false` items with commented out `= true` items, which is the same thing, really.

Reply:
> This is copied from strict=true; the idea is we can slowly work on making all these true.

OK, sounds good then, thanks!